Fetch from: https://github.com/nerves-project/nerves_system_br/blob/v1.20.4/patches/buildroot/0007-erlang-support-OTP-21-25.patch#L1050

From 6f015234f5a82d50fff05f944b3adc2d226174fc Mon Sep 17 00:00:00 2001
From: Frank Hunleth <fhunleth@troodon-software.com>
Date: Thu, 21 Mar 2019 20:49:54 -0400
Subject: [PATCH] erlang: enable deterministic builds

This adds the `+deterministic` compiler flag to the `erlc` calls to
strip out absolute paths in compiled `.beam` files.

Signed-off-by: Frank Hunleth <fhunleth@troodon-software.com>
---
 make/otp.mk.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/make/otp.mk.in b/make/otp.mk.in
index 4efffc0e25..f3ff11a011 100644
--- a/make/otp.mk.in
+++ b/make/otp.mk.in
@@ -110,6 +110,7 @@ ifeq ($(USE_ESOCK),yes)
 ERL_COMPILE_FLAGS += -DUSE_ESOCK=true
 endif

+ERL_COMPILE_FLAGS += +deterministic
 ERLC_WFLAGS = -W
 ERLC = erlc $(ERLC_WFLAGS) $(ERLC_FLAGS)
 ERL = erl -boot start_clean
